kbfssync/semaphore_test.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package kbfssync
+
+import (
+	"context"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+var testTimeout = 10 * time.Second
+
+type acquireCall struct {
+	n     int64
+	count int64
+	err   error
+}
+
+func callAcquire(ctx context.Context, s *Semaphore, n int64) acquireCall {
+	count, err := s.Acquire(ctx, n)
+	return acquireCall{n, count, err}
+}
+
+func requireNoCall(t *testing.T, callCh <-chan acquireCall) {
```

Comment:
If there were an actual bug, this would be a racy check since there's no guarantee the goroutines would have run when this called.  Ideally the semaphore would have a way to know when the caller is blocked, but I don't think that's required for this PR.  Maybe just a comment explaining what could be racy?